amazonka/src/Network/AWS/Internal/HTTP.hs

```diff
@@ -98,9 +98,9 @@ waiter w@Wait{..} x = do
 
     result rq = first (fromMaybe AcceptRetry . accept w rq) . join (,)
 
-    exit (AcceptSuccess, _) = return Nothing
```

Comment:
I'm not sure what case this was expected to handle, but I tried to preserve its semantics in spirit.